src/scripts/main.js

```diff
@@ -1,3 +1,204 @@
+
+  if (gameField[randomRow][randomColumn] === 0) {
+    gameField[randomRow][randomColumn] = Math.random() < 0.1 ? 4 : 2;
+  } else {
+    randomize();
+  }
+  updateCellStyles();
+}
+
+function gameStart() {
+  gameField = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  randomize();
+  randomize();
+  updateCellStyles();
+}
+
+function moveRight() {
+  gameField = gameField.map(row => {
+    return row.filter(num => num === 0)
+      .concat(row.filter(num => num !== 0));
+  });
+}
+
+function moveLeft() {
+  gameField = gameField.map(row => {
+    return row.filter(num => num !== 0)
+      .concat(row.filter(num => num === 0));
+  });
+}
+
+function moveDown() {
+  slideRight();
+  moveLeft();
+  slideLeft();
+}
+
+function moveUp() {
+  slideRight();
+  moveRight();
+  slideLeft();
+}
+
+function moveMix(move, combinate) {
+  move();
+  combinate();
+  move();
+}
+
+function slideRight() {
+  const newArray = [];
+
+  for (let x = 0; x < gameField[0].length; x++) {
+    newArray[x] = [];
+
+    for (let i = gameField.length - 1; i >= 0; i--) {
+      newArray[x].push(gameField[i][x]);
+    }
+  }
+
+  gameField = newArray;
+};
+
+function slideLeft() {
+  const newArray = [];
+
+  for (let x = 0; x < gameField[0].length; x++) {
+    newArray[x] = [];
+
+    for (let i = gameField.length - 1; i >= 0; i--) {
+      newArray[x].unshift(gameField[i][x]);
+    }
+  }
+
+  gameField = newArray.reverse();
+};
+
+function combineRow() {
+  for (const row of gameField) {
+    for (let i = 0; i < 4; i++) {
+      if (row[i] === row[i + 1]) {
+        const total = row[i] + row[i + 1];
+
+        row[i] = total;
+        row[i + 1] = 0;
+        score += total;
+        gameScore.innerHTML = score;
+      }
+    }
+  }
+}
+
+function combineColumn() {
+  slideRight();
+  combineRow();
+  slideLeft();
+}
+
+function updateCellStyles() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 4; col++) {
+      if (gameField[row][col] > 0) {
+        rowField[row].children[col].innerHTML = gameField[row][col];
+
+        rowField[row].children[col].className = `
+        field-cell field-cell--${gameField[row][col]}`;
+      } else {
+        rowField[row].children[col].innerHTML = '';
+        rowField[row].children[col].className = 'field-cell';
+      }
+    }
+  }
+}
+
+buttonStart.addEventListener('click', () => {
+  buttonStart.classList.replace('start', 'restart');
+  buttonStart.innerHTML = 'Restart';
+  messageStart.classList.add('hidden');
+  messageRestart.classList.remove('hidden');
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  gameScore.innerHTML = 0;
+  score = 0;
+
+  gameStart();
+});
+
+document.addEventListener('keyup', (key) => {
+  if (buttonStart.classList.contains('restart')) {
+    if (key.code === 'ArrowRight') {
+      moveMix(moveRight, combineRow);
+      randomize();
+    } else if (key.code === 'ArrowLeft') {
```

Comment:
It is better to use switch...case instead
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
Ok. I'll change to "switch case". Thanks for recommendation!)